### PR TITLE
Issue 333 panic when scanning wifi with unknown auth method

### PR DESCRIPTION
**Pull request by empirephoenix**
related https://github.com/esp-rs/embedded-svc/pull/65

### CODE REVIEW

File: src/wifi.rs [hunk 1]
```diff
@@ -194,22 +194,26 @@ impl TryFrom<&ClientConfiguration> for Newtype<wifi_sta_config_t> {
     }
 }
 
-impl From<Newtype<wifi_sta_config_t>> for ClientConfiguration {
+impl From<Newtype<wifi_sta_config_t>> for Option<ClientConfiguration> {
```
Comment (ivmarkov) [inline src/wifi.rs:197]: Why don't you keep the original code as-is, and only change
```rust
auth_method: Newtype(conf.0.threshold.authmode).into()
```

to
```rust
auth_method: Newtype(conf.0.threshold.authmode).into().unwrap()
```

?

File: src/wifi.rs [hunk 2]
```diff
@@ -237,25 +241,32 @@ impl TryFrom<&AccessPointConfiguration> for Newtype<wifi_ap_config_t> {
     }
 }
 
-impl From<Newtype<wifi_ap_config_t>> for AccessPointConfiguration {
+impl From<Newtype<wifi_ap_config_t>> for Option<AccessPointConfiguration> {
```
Comment (ivmarkov) [inline src/wifi.rs:241]: Ditto.

File: src/wifi.rs [hunk 3]
```diff
@@ -1127,8 +1138,12 @@ impl<'d> WifiDriver<'d> {
         let mut wifi_config: wifi_config_t = Default::default();
         esp!(unsafe { esp_wifi_get_config(wifi_interface_t_WIFI_IF_STA, &mut wifi_config) })?;
 
-        let result: ClientConfiguration = unsafe { Newtype(wifi_config.sta).into() };
-
+        let client_config: Option<ClientConfiguration>;
```
Comment (ivmarkov) [inline src/wifi.rs:1133]: This would then stay as-is.

File: src/wifi.rs [hunk 4]
```diff
@@ -1158,8 +1173,12 @@ impl<'d> WifiDriver<'d> {
...
         let mut wifi_config: wifi_config_t = Default::default();
         esp!(unsafe { esp_wifi_get_config(wifi_interface_t_WIFI_IF_AP, &mut wifi_config) })?;
-
-        let result: AccessPointConfiguration = unsafe { Newtype(wifi_config.ap).into() };
+        let access_point_configuration: Option<AccessPointConfiguration>;
+        unsafe {
+            access_point_configuration = Newtype(wifi_config.ap).into();
+        }
+        //unwrap should be safe here, as it should be impossible to configure a accesspoint, that cannot be represented
+        let result: AccessPointConfiguration = unsafe { access_point_configuration.unwrap() };
```
Comment (ivmarkov) [inline src/wifi.rs]: Why the `unsafe` here? There is nothing `unsafe` in unwrapping an `Option`. It might panic, but that's all.

Comment (ivmarkov) [reply src/wifi.rs]: Anyway, here also this change would be unnecessary, once you restore the original `From` definition for `AccessPointConfiguration`.

**Review by ivmarkov — CHANGES_REQUESTED** [review]

File: src/wifi.rs (hunk 1 above)
Comment (empirephoenix) [reply src/wifi.rs:197]: Hm actually, yeah good point..., let me adjust stuff

**Comment by ivmarkov** [comment]
Thanks!
CI fails ATM due to unrelated issues in Rust STD. hopefully to be addressed beginning of next week.

**Pull request merged**